### DISCUSSION TITLE
Stopped Working @ 2:36 AM :: Traceback included

### DISCUSSION THREAD

**Discussion by gniknalu** [roflcoopter/viseron#318] [Q&A] [answered]
I have no idea what happened. I can't decipher, from the logs, what is wrong other than to guess that something Python related must have updated last night killing viseron.

The logs don't tell me the story - hopefully you can help.

If I leave it running, it just restarts over and over with the same errors.

Running the dev version

> docker logs -f viseron
> [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
> [s6-init] ensuring user provided files have correct perms...[cont-finish.d] executing container finish scripts...
> [cont-finish.d] done.
> [s6-finish] waiting for services.
> exited 0.
> [fix-attrs.d] applying ownership & permissions fixes...
> [fix-attrs.d] done.
> [cont-init.d] executing container initialization scripts...
> [cont-init.d] 10-adduser: executing...
> [s6-finish] sending all processes the TERM signal.
> [s6-finish] sending all processes the KILL signal and exiting.
> root@fastNVR:/usr/share/viseron# docker logs -f viseron
> [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
> [s6-init] ensuring user provided files have correct perms...[cont-finish.d] executing container finish scripts...
> [cont-finish.d] done.
> [s6-finish] waiting for services.
> exited 0.
> [fix-attrs.d] applying ownership & permissions fixes...
> [fix-attrs.d] done.
> [cont-init.d] executing container initialization scripts...
> [cont-init.d] 10-adduser: executing...
> [s6-finish] sending all processes the TERM signal.
> [s6-finish] sending all processes the KILL signal and exiting.
> [s6-init] making user provided files available at /var/run/s6/etc...exited 0.
> [s6-init] ensuring user provided files have correct perms...exited 0.
> [fix-attrs.d] applying ownership & permissions fixes...
> [fix-attrs.d] done.
> [cont-init.d] executing container initialization scripts...
> [cont-init.d] 10-adduser: executing...
> ************************ UID/GID *************************
> User uid:    0
> User gid:    0
> ************************** Done **************************
> [cont-init.d] 10-adduser: exited 0.
> [cont-init.d] 20-gid-video-device: executing...
> [cont-init.d] 20-gid-video-device: exited 0.
> [cont-init.d] 30-edgetpu-permission: executing...
> ************** Setting EdgeTPU permissions ***************
> Coral Vendor IDs:
> "1a6e",ATTRS{idProduct}=="089a"
> "18d1",ATTRS{idProduct}=="9302"
> No EdgeTPU USB device was found
> ************************** Done **************************
> [cont-init.d] 30-edgetpu-permission: exited 0.
> [cont-init.d] 40-set-env-vars: executing...
> ****** Checking for hardware acceleration platforms ******
> OpenCL is available!
> VA-API cannot be used
> CUDA is available!
> *********************** Done *****************************
> [cont-init.d] 40-set-env-vars: exited 0.
> [cont-init.d] 50-check-if-rpi: executing...
> ********** Checking if we are running on an RPi **********
> Not running on any supported RPi
> *********************** Done *****************************
> [cont-init.d] 50-check-if-rpi: exited 0.
> [cont-init.d] 55-check-if-jetson: executing...
> ****** Checking if we are running on a Jetson Board ******
> Not running on any supported Jetson board
> *********************** Done *****************************
> [cont-init.d] 55-check-if-jetson: exited 0.
> [cont-init.d] 60-ffmpeg-path: executing...
> ****************** Getting FFmpeg path *******************
> FFmpeg path: /usr/local/bin/ffmpeg
> *********************** Done *****************************
> [cont-init.d] 60-ffmpeg-path: exited 0.
> [cont-init.d] done.
> [services.d] starting services
> [services.d] done.
> [2022-02-09 21:37:25] [INFO    ] [viseron                 ] - -------------------------------------------
> [2022-02-09 21:37:25] [INFO    ] [viseron                 ] - Initializing...
> [2022-02-09 21:37:26] [INFO    ] [viseron.mqtt            ] - Initializing MQTT connection
> [2022-02-09 21:37:27] [INFO    ] [viseron                 ] - Initializing NVR threads
> Exception in thread setup.BackEast:
> Traceback (most recent call last):
>   File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
> Exception in thread Exception in thread setup.FrontWestsetup.FrontEast2Exception in thread :
> :
> Traceback (most recent call last):
>   File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
>     self.run()
>   File "/src/viseron/__init__.py", line 134, in run
>     camera_config = NVRConfig(
>   File "/src/viseron/config/__init__.py", line 202, in __init__
>     self._camera = CameraConfig(camera, motion_detection)
>   File "/src/viseron/config/config_camera.py", line 385, in __init__
>     self._validated_config = self._schema(camera)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
>     return self._compiled([], data)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 205, in _run
>     return self._exec(self._compiled, value, path)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 287, in _exec
>     raise e if self.msg is None else AllInvalid(self.msg, path=path)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 285, in _exec
>     v = func(path, v)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 817, in validate_callable
>     return schema(data)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
>     return self._compiled([], data)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 594, in validate_dict
>     return base_validate(path, iteritems(data), out)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 432, in validate_mapping
>     raise er.MultipleInvalid(errors)
> voluptuous.error.MultipleInvalid: extra keys not allowed @ data['motion_detection']['interval']
> setup.BackWestTraceback (most recent call last):
> :
> Traceback (most recent call last):
>       File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
>   File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
> self.run()
>   File "/src/viseron/__init__.py", line 134, in run
>         self.run()
>     self.run()
>   File "/src/viseron/__init__.py", line 134, in run
>   File "/src/viseron/__init__.py", line 134, in run
> camera_config = NVRConfig(
>   File "/src/viseron/config/__init__.py", line 202, in __init__
>         camera_config = NVRConfig(self._camera = CameraConfig(camera, motion_detection)
> 
>   File "/src/viseron/config/config_camera.py", line 385, in __init__
>   File "/src/viseron/config/__init__.py", line 202, in __init__
>         self._camera = CameraConfig(camera, motion_detection)
> camera_config = NVRConfig(  File "/src/viseron/config/config_camera.py", line 385, in __init__
> 
>   File "/src/viseron/config/__init__.py", line 202, in __init__
> self._validated_config = self._schema(camera)    self._validated_config = self._schema(camera)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
> 
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
>     return self._compiled([], data)    self._camera = CameraConfig(camera, motion_detection)
>   File "/src/viseron/config/config_camera.py", line 385, in __init__
> 
>     self._validated_config = self._schema(camera)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 205, in _run
>         return self._exec(self._compiled, value, path)
>       File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 287, in _exec
> return self._compiled([], data)return self._compiled([], data)
> 
>     raise e if self.msg is None else AllInvalid(self.msg, path=path)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 285, in _exec
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 205, in _run
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 205, in _run
>     v = func(path, v)
>         return self._exec(self._compiled, value, path)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 287, in _exec
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 817, in validate_callable
> return self._exec(self._compiled, value, path)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 287, in _exec
>     raise e if self.msg is None else AllInvalid(self.msg, path=path)    raise e if self.msg is None else AllInvalid(self.msg, path=path)
> 
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 285, in _exec
> return schema(data)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/validators.py", line 285, in _exec
>     v = func(path, v)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 817, in validate_callable
>     return self._compiled([], data)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 594, in validate_dict
>     v = func(path, v)    return base_validate(path, iteritems(data), out)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 432, in validate_mapping
>     return schema(data)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
> 
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 817, in validate_callable
> return self._compiled([], data)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 594, in validate_dict
>     return schema(data)
>       File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 272, in __call__
> raise er.MultipleInvalid(errors)
> return base_validate(path, iteritems(data), out)voluptuous.error.MultipleInvalid:     return self._compiled([], data)extra keys not allowed @ data['motion_detection']['interval']
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 594, in validate_dict
> 
> 
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 432, in validate_mapping
>     return base_validate(path, iteritems(data), out)
>   File "/usr/local/lib/python3.8/dist-packages/voluptuous/schema_builder.py", line 432, in validate_mapping
>         raise er.MultipleInvalid(errors)
> voluptuous.errorraise er.MultipleInvalid(errors)
> .MultipleInvalidvoluptuous.error: extra keys not allowed @ data['motion_detection']['interval']
> .MultipleInvalid: extra keys not allowed @ data['motion_detection']['interval']
> [2022-02-09 21:37:27] [INFO    ] [viseron                 ] - Initialization complete
> [cont-finish.d] executing container finish scripts...
> [cont-finish.d] done.
> [s6-finish] waiting for services.
> [s6-finish] sending all processes the TERM signal.
> [s6-finish] sending all processes the KILL signal and exiting.

**Comment by roflcoopter** [comment]
Thats odd, are you on `dev`?

**Reply by gniknalu** [reply]
> Thats odd, are you on `dev`?

Yes - on dev.

I've tested the CUDA stuff and it's working and also have rebooted a couple of times just in case. . . . .

**Reply by gniknalu** [reply]
> Thats odd, are you on `dev`?

I've even removed all but one camera and get the same thing - restarting over and over wit the same errors for just the one camera.

**Reply by roflcoopter** [reply]
Ahh then its probably because it pulled a new version of `dev`
A while back `interval` changed to `fps` instead

**Reply by gniknalu** [reply]
> Ahh then its probably because it pulled a new version of `dev` A while back `interval` changed to `fps` instead

That seemed to fix it - LOL - **but you'll never guess what I'm seeing**? The same issue from the past where the detection from one camera is cloned to the other cameras.

**Reply by roflcoopter** [reply]
hahah FML....

**Reply by roflcoopter** [reply]
Been a while since i touched `dev` so not sure if any changes were made there.
Maybe you could move to `latest` until `v2` is done.

I had this problem there as well but i made sure to fix it for good there

**Reply by gniknalu** [reply]
> Been a while since i touched `dev` so not sure if any changes were made there. Maybe you could move to `latest` until `v2` is done.
> 
> I had this problem there as well but i made sure to fix it for good there

Sure - doing that right now. Thank you

**Comment by gniknalu** [comment, marked as answer]
Switching to the modularized version.
